### PR TITLE
Refactor location lock

### LINKED ISSUE #383: [Main screen] Try to enable location lock when the app is opened
Currently, there viewport is not explicitly defined when the app is opened. We assume users will be collecting data in situ, so it generally makes sense to pan/zoom the map to their current location.

Note: the project selector will be moved to its own fragment in which the map will not be shown, so for now it's ok to show attempt location lock as soon as the HomeScreenFragment is shown.

### PR DESCRIPTION
**Pull request by shobhitagarwal1612**
Closes #383

Things done:

- [x] Create a shared layout for location lock 
- [x] Create a shared view model for location lock
- [x] Cleanup existing usages. This also ensures that the state is maintained across screens
- [x] Trigger location lock toggle when app starts.
- [x] Create a module for LocaionManager. Then uninstall the module for unit test and replace with a mock

### CODE REVIEW

File: gnd/src/main/res/layout/location_lock_btn.xml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    <variable
+      name="viewModel"
+      type="com.google.android.gnd.ui.home.LocationLockViewModel" />
+  </data>
+
+  <com.google.android.material.floatingactionbutton.FloatingActionButton
+    android:id="@+id/location_lock_btn"
+    android:layout_width="wrap_content"
+    android:layout_height="wrap_content"
+    android:enabled="@{viewModel.locationLockEnabled}"
```
Comment (JSunde) [inline gnd/src/main/res/layout/location_lock_btn.xml:32]: Huh, I don't think I've seen this syntax in XML before, is that Kotlin?

File: gnd/src/main/java/com/google/android/gnd/ui/home/LocationLockViewModel.kt [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+import com.google.android.gnd.ui.common.SharedViewModel
+import com.google.android.gnd.util.toLiveData
+import io.reactivex.BackpressureStrategy
+import io.reactivex.Flowable
+import io.reactivex.subjects.PublishSubject
+import io.reactivex.subjects.Subject
+import javax.inject.Inject
+
+@SharedViewModel
+class LocationLockViewModel @Inject constructor(private val locationManager: LocationManager) :
```
Comment (JSunde) [inline gnd/src/main/java/com/google/android/gnd/ui/home/LocationLockViewModel.kt:36]: Could you add tests for this file?

**Review by JSunde — CHANGES_REQUESTED** [review]

**Comment by gino-m** [comment]
> @... PTAL?

Nit: Please remember to update/remove PR description template strings before sending :)

File: gnd/src/main/res/layout/location_lock_btn.xml (hunk 1 above)
Comment (shobhitagarwal1612) [reply gnd/src/main/res/layout/location_lock_btn.xml:32]: This is data binding. (not related to Kotlin)

https://developer.android.com/topic/libraries/data-binding

File: gnd/src/main/java/com/google/android/gnd/ui/home/LocationLockViewModel.kt (hunk 2 above)
Comment (shobhitagarwal1612) [reply gnd/src/main/java/com/google/android/gnd/ui/home/LocationLockViewModel.kt:36]: Done

File: gnd/src/main/res/layout/location_lock_btn.xml (hunk 1 above)
Comment (JSunde) [reply gnd/src/main/res/layout/location_lock_btn.xml:32]: Gotcha, thanks for the info!

**Review by JSunde — APPROVED** [review]

**Comment by shobhitagarwal1612** [comment]
Adding coarse permissions also didn't help.

https://alexzh.com/ui-testing-of-android-runtime-permissions/

**Comment by shobhitagarwal1612** [comment]
Closing this PR for now

**Pull request closed without merging**